src/linker/utilities/data_utils.py

```diff
@@ -1,42 +1,24 @@
 import os
-import shutil
 from datetime import datetime
 from pathlib import Path
-from typing import TYPE_CHECKING, Dict, List, Optional
-
-if TYPE_CHECKING:
-    from linker.configuration import Config
+from typing import Dict, Optional, Union
 
 import yaml
 
 
-def create_results_directory(output_dir: Optional[str], timestamp: bool) -> Path:
+def get_results_directory(output_dir: Optional[str], timestamp: bool) -> Path:
```

Comment:
I thought it would be better to have the config make these directories as part of the copy process that happens after validation.